### PR TITLE
Fix WebDatasets KeyError for user-defined Features when a field is missing in an example

### PR DESCRIPTION
**Pull request by ProGamerGov**
Fixes: https://github.com/huggingface/datasets/issues/6900

Not sure if this needs any addition stuff before merging

### CODE REVIEW

File: src/datasets/packaged_modules/webdataset/webdataset.py [hunk 1]
```diff
@@ -109,8 +109,12 @@ def _generate_examples(self, tar_paths, tar_iterators):
...
             field_name for field_name, feature in self.info.features.items() if isinstance(feature, datasets.Audio)
         ]
+        all_field_names = list(self.config.features.keys())
         for tar_idx, (tar_path, tar_iterator) in enumerate(zip(tar_paths, tar_iterators)):
             for example_idx, example in enumerate(self._get_pipeline_from_tar(tar_path, tar_iterator)):
+                for field_name in all_field_names:
+                    if field_name not in example:
+                        example[field_name] = None
                 for field_name in image_field_names + audio_field_names:
                     example[field_name] = {"path": example["__key__"] + "." + field_name, "bytes": example[field_name]}
```
Comment (lhoestq) [inline src/datasets/packaged_modules/webdataset/webdataset.py:119]:
```suggestion
                    if example[field_name] is not None:
                        example[field_name] = {"path": example["__key__"] + "." + field_name, "bytes": example[field_name]}
```

**Review by lhoestq — COMMENTED** [review]
Great ! I added a small suggestion to handle the case where the missing field is an image or an audio sample.

Could you also add a small test in `test_webdataset.py` ?
I believe you can reuse the same code as in `test_image_webdataset` and simply pass `features=` to `Webdataset()` and check if the examples have the extra features as None. Then we'll be good to merge !

**Review by lhoestq — APPROVED** [review]
Great ! I updated the test :) the rest of the CI failures are unrelated to this PR

**Comment by ProGamerGov** [comment]
@lhoestq Thank you!

**Pull request merged**